nifi-nar-bundles/nifi-windows-event-log-bundle/nifi-windows-event-log-processors/src/main/java/org/apache/nifi/processors/windows/event/log/ConsumeWindowsEventLog.java

```diff
@@ -199,6 +230,8 @@ private String subscribe(ProcessContext context) throws URISyntaxException {
 
         subscriptionHandle = wEvtApi.EvtSubscribe(null, null, channel, query, null, null,
                 evtSubscribeCallback, WEvtApi.EvtSubscribeFlags.SUBSCRIBE_TO_FUTURE | WEvtApi.EvtSubscribeFlags.EVT_SUBSCRIBE_STRICT);
+        lastActivityTimestamp = System.currentTimeMillis();
```

Comment:
Just wanted to clarify the logic here... Do we want to update the `lastActivityTimestamp` when the client was unable to subscribe? Could that lead to premature reconnection later?

Reply:
I think if the client failed to subscribe, it keeps trying reconnecting at subsequent onTrigger. So, updating lastActivityTimestamp in case of subscription failure will not lead to premature I believe. But, I updated the code anyway. It looks more correct in that way, thanks!